Ask for confirmation to delete an expense

Gukkey opened the pull request:
Resolves #102 

![image](https://github.com/spliit-app/spliit/assets/74915094/3bee0579-33a7-44e5-9950-17b960659818)

Right now I have created a separate component for the popup which is named "delete-popup". Let me know if I have to create a more generic component first (something like "popup"). Let me also know if I have to customize the popup further in any way.

justcallmelarry commented:
I think this is a great addition!

I don't really have anything to say on the best practice use of components, so I'll let someone else have a say about those.

I did want to say that at least the formatting checks are going to fail, though, so you might want to run prettier before Sebastien has a look at anything (`npx prettier -w .`) in order to fix the formatting

Gukkey commented:
I have run my files through prettier now (thanks for your comment!), and instead of just using `<DialogClose>` I will add a `<Button>` inside of it (I think it would make the close option look like a proper button and they would have same font size) and make a PR within today. Thanks for your input.

Gukkey commented:
I have fixed the formatting issues and the cancel option is now a proper button now. Also, I have changed the title and description a little. 

![image](https://github.com/spliit-app/spliit/assets/74915094/636809ec-5973-4fd2-a292-165183c999d4)

shynst commented on src/components/delete-popup.tsx:
```diff
@@ -0,0 +1,51 @@
+          Delete
+        </Button>
+      </DialogTrigger>
+      <DialogPortal>
```
You probably should remove the enclosing `<DialogPortal>` as well as the following `<DialogOverlay />`, since both elements are already included in `<DialogContent />`

So basically you are adding them twice. Just having one `<DialogOverlay />` also makes the background a little bit transparent which looks better, too.

shynst reviewed (changes requested):
Looks fine. Just one little remark, though...

shynst commented on src/components/delete-popup.tsx:
```diff
@@ -0,0 +1,51 @@
+            Do you really want to delete this expense? This action is
+            irreversible.
+          </DialogDescription>
+          <div className="flex gap-2">
```
Try `<DialogFooter>`instead of this `<div>`

shynst commented on src/components/delete-popup.tsx:
```diff
@@ -0,0 +1,51 @@
+            </AsyncButton>
+            <DialogClose>
+              <Button variant={'secondary'}>Cancel</Button>
+            </DialogClose>
```
Looks better on on small screens, if these two are reversed - i.e. `<Button><DialogClose>Cancel</DialogClose></Button>`

shynst commented on src/components/delete-popup.tsx:
```diff
@@ -0,0 +1,51 @@
+      </DialogTrigger>
+      <DialogPortal>
+        <DialogOverlay />
+        <DialogContent>
```
You probably should remove the enclosing `<DialogPortal>` as well as the following `<DialogOverlay />`, since both elements are already included in `<DialogContent />`.

So basically you are adding them twice. Just having one `<DialogOverlay />` also makes the background a little bit transparent which looks better, too.

shynst reviewed (changes requested):
Looks good! Just some minor UI related stuff...

Gukkey commented:
Hi @shynst, thanks for your input. I have pushed a commit incorporating your suggestions.

shynst reviewed (approved):
Great! Looks fine to me.

scastiel commented:
Great, thanks 👏
